{{cookiecutter.project_slug}}/clients/web/vue3/src/services/users/forms.js

```diff
@@ -33,4 +33,10 @@ export class PasswordResetForm extends Form {
       new MinLengthValidator({ minLength: 8, message: 'Minimum Length of 8 required' }),
     ],
   })
+  static passwordConfirmation = new FormField({
+    validators: [
+      new RequiredValidator(),
+      new MinLengthValidator({ minLength: 8, message: 'Minimum Length of 8 required' }),
+    ],
+  })
 }
```

Comment:
For form level validators there is a special key (eventually need to make it a symbol)
```
export class PasswordResetForm extend Form{
 ....fields....
    // add this 
   static dynamicFormValidators = {
    confirmPassword: [new MustMatchValidator({ matcher: 'password' })],
  }
  }
}
```

Reply:
This (and the tn-forms docs) confuse me a little here. Given that this is password confirmation, shouldn't I just drop the two existing validators (redundant) and just add the MustMatchValidator? Or is the whole dynamicFormValidators step somehow special?

Reply:
I tried the idea I stated above and things are now broken. It always says the passwords don't match even when they do.

Reply:
yea it has to be set up with the dynamicFormValidators instead this is the case for all FormLevel validators. The reason is that we cannot get the reference to the value until the form is created (i.e the same pointer in memory). The dynamic form validator object is set after we create the form and feeds the validator the form field.

Reply:
but you are right you can just drop the two existing ones if you want I add them just as an additional measure to remind the user to fill them in.

Reply:
@oudeismetis did you want me to complete this for you?

Reply:
yes please